compiler/shared/test/scala/mlscript/compiler/TestIR.scala

```diff
@@ -6,18 +6,40 @@ import mlscript.compiler.ir._
 import scala.collection.mutable.StringBuilder
 import mlscript.{DiffTests, ModeType, TypingUnit}
 import mlscript.compiler.ir.{Interpreter, Fresh, FreshInt, Builder}
+import mlscript.compiler.optimizer.TailRecOpt
+import mlscript.Diagnostic
 
 class IRDiffTestCompiler extends DiffTests {
   import IRDiffTestCompiler.*
 
-  override def postProcess(mode: ModeType, basePath: List[Str], testName: Str, unit: TypingUnit, output: Str => Unit): (List[Str], Option[TypingUnit]) = 
+  override def postProcess(mode: ModeType, basePath: List[Str], testName: Str, unit: TypingUnit, output: Str => Unit, raise: Diagnostic => Unit): (List[Str], Option[TypingUnit]) = 
     val outputBuilder = StringBuilder()
     if (mode.useIR || mode.irVerbose)
       try
         output("\n\nIR:")
-        val gb = Builder(Fresh(), FreshInt(),  FreshInt(), FreshInt())
-        val graph = gb.buildGraph(unit)
-        output(graph.toString())
+        val fnUid = FreshInt()
+        val classUid = FreshInt()
+        val tag = FreshInt()
+
+        val gb = Builder(Fresh(), fnUid, classUid, tag, raise)
+        val graph_ = gb.buildGraph(unit)
+        
+        if !mode.noTailRecOpt then 
+          output(graph_.toString())
+
+        val graph = if (!mode.noTailRecOpt) {
+          val tailRecOpt = new TailRecOpt(fnUid, classUid, tag, raise)
+          val (g, comps) = tailRecOpt.run_debug(graph_)
+          output("\nStrongly Connected Tail Calls:")
+          output(comps.toString)
+          g
+        } else {
+          graph_
+        }
```

Comment:
This subproject is on Scala 3, isn't it? It's more idiomatic, elegant, and easier to edit to make it:
```suggestion
        val graph = if !mode.noTailRecOpt
          then
            val tailRecOpt = new TailRecOpt(fnUid, classUid, tag, raise)
            val (g, comps) = tailRecOpt.run_debug(graph_)
            output("\nStrongly Connected Tail Calls:")
            output(comps.toString)
            g
          else
            graph_
```